usd/src/sdf2usd_TEST.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <ignition/common/TempDirectory.hh>
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"
+#include "test_utils.hh"
+
+#ifdef _WIN32
+  #define popen  _popen
+  #define pclose _pclose
+#endif
+
+static std::string sdf2usdCommand()
+{
+  return std::string(IGN_PATH) + "/sdf2usd";
+}
+
+/////////////////////////////////////////////////
+std::string custom_exec_str(std::string _cmd)
+{
+  _cmd += " 2>&1";
+  FILE *pipe = popen(_cmd.c_str(), "r");
+
+  if (!pipe)
+    return "ERROR";
+
+  char buffer[128];
+  std::string result = "";
+
+  while (!feof(pipe))
+  {
+    if (fgets(buffer, 128, pipe) != NULL)
+      result += buffer;
+  }
+
+  pclose(pipe);
+  return result;
+}
+
+/////////////////////////////////////////////////
+TEST(check_cmd, IGN_UTILS_TEST_DISABLED_ON_WIN32(SDF))
+{
+  std::string pathBase = PROJECT_SOURCE_PATH;
+  pathBase = ignition::common::joinPaths(pathBase, "test", "sdf");
+
+  auto tmpDir = ignition::common::tempDirectoryPath();
+  auto tmp = ignition::common::createTempDirectory("usd", tmpDir);
+  // Check a good SDF file
+  {
+    std::string path = ignition::common::joinPaths(pathBase,
+      "/shapes_world.sdf");
+    std::string output =
+      custom_exec_str(sdf2usdCommand() + " " + path + " " +
+        ignition::common::joinPaths(tmp, "shapes.usd"));
+    // TODO(ahcorde): Check the output when the parser is implemented
+  }
```

Comment:
Should we add a check to make sure that a `shapes.usd` file was actually generated? I imagine that we can check for file existence with this method: https://github.com/ignitionrobotics/ign-common/blob/ignition-common4_4.5.0/include/ignition/common/Filesystem.hh#L55-L58

Reply:
Right now,  I believe the file is not generated

Reply:
If I build this PR locally, I can use the `sdf2usd` executable (it's generated), so is there a reason why it wouldn't be generated/available in the test?

Reply:
did you check if `shapes.usd` is generated ?

Reply:
Ah, sorry, I misunderstood your previous comment - I thought you said the `sdf2usd` executable isn't generated. Here's what I did to test if `shapes.usd` is generated:

```diff
diff --git a/usd/src/sdf2usd_TEST.cc b/usd/src/sdf2usd_TEST.cc
index b929b1f8..0f02f617 100644
--- a/usd/src/sdf2usd_TEST.cc
+++ b/usd/src/sdf2usd_TEST.cc
@@ -18,6 +18,7 @@
 
 #include <gtest/gtest.h>
 
+#include <ignition/common/Filesystem.hh>
 #include <ignition/common/TempDirectory.hh>
 #include <ignition/utilities/ExtraTestMacros.hh>
 
@@ -71,6 +72,8 @@ TEST(check_cmd, IGN_UTILS_TEST_DISABLED_ON_WIN32(SDF))
     std::string output =
       custom_exec_str(sdf2usdCommand() + " " + path + " " +
         ignition::common::joinPaths(tmp, "shapes.usd"));
+    // make sure that a shapes.usd file was generated
+    EXPECT_TRUE(ignition::common::isFile("shapes.usd"));
     // TODO(ahcorde): Check the output when the parser is implemented
   }
 }
```

If I run the test, the check fails. I think it's not being generated because we aren't searching in the correct location for the `sdf2usd` executable, which I mentioned in a different comment thread: https://github.com/ignitionrobotics/sdformat/pull/817#discussion_r788125682

Reply:
@adlarkin Done here https://github.com/ignitionrobotics/sdformat/pull/817/commits/0195564045cfc260663d462f4418590b29de6f45